src/reactviews/pages/QueryResult/table/plugins/headerFilter.plugin.ts

```diff
@@ -183,25 +216,81 @@ export class HeaderFilter<T extends Slick.SlickData> {
         // Set the new popup as the active popup
         this.activePopup = $popup;
         const checkboxContainer = $popup.find("#checkbox-list");
-        if (this.listData) {
-            this.listData.forEach((option) => {
-                const checkbox = jQuery(
-                    `<label><input type="checkbox" value="${option.value}"> ${option.displayText}</label>`,
+        this._list = new VirtualizedList(
+            checkboxContainer.get(0),
+            this._listData,
+            (itemContainer, item) => {
```

Comment:
Can you pop this long function and the next one out into named local functions so that this constructor invocation is easier to read?